jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -1057,7 +1057,7 @@ public void earlyEOF()
             HttpStreamOverHTTP1 stream = _stream.get();
             if (stream != null)
             {
-                BadMessageException bad = new BadMessageException("Early EOF");
+                EofException bad = new EofException("Early EOF");
```

Comment:
I don't like that this sends a 500.    I accept that perhaps the request is not bad and it is a connection problem, but it is certainly not a server problem as the 500 suggests.
 How about:
```suggestion
                BadMessageException bad = new BadMessageException(null, new EofException("Early EOF"));
```

Reply:
I think it should be a type of EOF exception because that's what it is.
I have introduced an `HttpEofException` so it is converted to a 400, but it is-a `IOException`, and it's quiet because it's a client close, so won't clutter the logs.